src/services/shardQuerySplitting.ts

```diff
@@ -129,13 +129,13 @@ function splitQueriesByStreamShard(
             return;
           }
         }
-        if (mergedResponse.data.length) {
-          mergedResponse.state = LoadingState.Streaming;
-        }
         mergedResponse = combineResponses(mergedResponse, partialResponse);
       },
       complete: () => {
-        subscriber.next(mergedResponse);
+        // Prevent flashing "no data"
+        if (mergedResponse.data.length) {
```

Comment:
What if we only ever get empty frames. I think we need to at least call `subscriber.next(mergedResponse);` when the `state` is `LoadingState.Done` or something like that?

Reply:
Good question. The done() function takes care of that. https://github.com/grafana/explore-logs/blob/main/src/services/shardQuerySplitting.ts#L77